diff --git a/configure b/configure
index 87b65d86d1..3bef023e08 100755
--- a/configure
+++ b/configure
@@ -6596,7 +6596,22 @@ enabled libmodplug        && require_pkg_config libmodplug libmodplug libmodplug
 enabled libmp3lame        && require "libmp3lame >= 3.98.3" lame/lame.h lame_set_VBR_quality -lmp3lame $libm_extralibs
 enabled libmysofa         && { check_pkg_config libmysofa libmysofa mysofa.h mysofa_neighborhood_init_withstepdefine ||
                                require libmysofa mysofa.h mysofa_neighborhood_init_withstepdefine -lmysofa $zlib_extralibs; }
-enabled libnpp            && { check_lib libnpp npp.h nppGetLibVersion -lnppig -lnppicc -lnppc -lnppidei -lnppif ||
+enabled libnpp            && { check_lib libnpp npp.h nppGetLibVersion \
+                                 -lnppig_static -lnppicc_static -lnppif_static -lnppim_static \
+                                 -lnppitc_static -lnppial_static -lnppisu_static -lnppist_static \
+                                 -lnppidei_static -lnpps_static -lnppc_static \
+                                 -lcudart_static -lculibos $ldl ||
+                               check_lib libnpp npp.h nppGetLibVersion \
+                                 -lnppig_static -lnppicc_static -lnppif_static -lnppim_static \
+                                 -lnppitc_static -lnppial_static -lnppisu_static -lnppist_static \
+                                 -lnppidei_static -lnpps_static -lnppc_static \
+                                 -lcudart_static -lculibos $ldl -lstdc++ ||
+                               check_lib libnpp npp.h nppGetLibVersion \
+                                 -lnppig_static -lnppicc_static -lnppif_static -lnppim_static \
+                                 -lnppitc_static -lnppial_static -lnppisu_static -lnppist_static \
+                                 -lnppidei_static -lnpps_static -lnppc_static \
+                                 -lcudart_static -lculibos $ldl -lstdc++ -lrt \
+                               check_lib libnpp npp.h nppGetLibVersion -lnppig -lnppicc -lnppc -lnppidei -lnppif ||
                                check_lib libnpp npp.h nppGetLibVersion -lnppi -lnppif -lnppc -lnppidei ||
                                die "ERROR: libnpp not found"; }
 enabled libopencore_amrnb && require libopencore_amrnb opencore-amrnb/interf_dec.h Decoder_Interface_init -lopencore-amrnb
